watcher/watcher.go

```diff
@@ -179,9 +189,104 @@ func (wh *WatchHandler) getImagesIDsToWlidMap() map[string][]string {
 	return wh.imagesIDToWlidsMap
 }
 
+// HandleSBOMEvents handles SBOM-related events
+//
+// Handling events is defined as dispatching scan commands that match a given
+// SBOM to the main command channel
+func (wh *WatchHandler) HandleSBOMEvents(sbomEvents <-chan watch.Event, commands chan<- *apis.Command, errorCh chan<- error) {
+	defer close(commands)
+	defer close(errorCh)
+
+	for event := range sbomEvents {
+		obj, ok := event.Object.(*spdxv1beta1.SBOMSPDXv2p3)
+		if !ok {
+			errorCh <- ErrUnsupportedObject
+			continue
+		}
+
+		// We need to schedule scans only for new SBOMs, not changes to
+		// existing ones or other operations
+		if event.Type != watch.Added {
+			continue
+		}
+
+		imageID := obj.ObjectMeta.Name
+
+		wh.imageIDsMapMutex.RLock()
+		wlids, ok := wh.imagesIDToWlidsMap[imageID]
+		wh.imageIDsMapMutex.RUnlock()
```

Comment:
thought: we should really abstract the image IDs to WLIDs map into a separate struct

This was the reason behind many nasty deadlocks I have encountered, especially with how often we like to use `defer mu.RUnlock()`.

Another note here is that, in my opinion, Pod, SBOM and SBOM Filtered watches should be separate structs that share the image ID to WLIDs map or some other map they have. Because so far it seems that the `WatchHandler` struct does too many things.